stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api/HeadersInterceptor.kt

```diff
@@ -4,9 +4,9 @@ import io.getstream.chat.android.client.ChatClient
 import okhttp3.Interceptor
 import okhttp3.Response
 
-internal class HeadersInterceptor(val config: ChatClientConfig) : Interceptor {
+internal class HeadersInterceptor(private val isAnonymous: () -> Boolean) : Interceptor {
```

Comment:
Why do you need lambda here instead of simple boolean flag?

Reply:
Yep, the value changes when we authenticate with "anonymous" or "normal"/"guest" users

Reply:
But you provide a lambda as closure without any state (variable) I can't see that it changes when we authenticated. Or it will be done later in another PR?

Reply:
The idea in general: 
- For "getGuestToken" server call we have a provider which always returns true
- For every other method we pass `{ config.isAnonymous }` so that `isAnonymous: () -> Boolean` lambda returns different value depending on the auth state:
1) Not authenticated -> `isAnonymous()` should return false
2) Anonymous auth -> `isAnonymous()` should return true
3) Regular/guest auth -> `isAnonymous()` should return true

Reply:
The value of `config.isAnonymous` changes when we set a user:
```
    override fun setAnonymousUser(listener: InitConnectionListener?) {
        config.isAnonymous = true
         ...
    }

```

```
    override fun setUser(user: User, token: String, listener: InitConnectionListener?) {
        config.isAnonymous = false
         ...
    }

```

Reply:
Now, I see. Thanks for clarification